src/lints/pub_function_changed_abi.ron

```diff
@@ -0,0 +1,58 @@
+SemverQuery(
+    id: "pub_function_changed_abi",
+    human_readable_name: "pub function changed ABI",
+    description: "A public function changed its external ABI.",
+    required_update: Major,
+    reference_link: Some("https://doc.rust-lang.org/reference/items/external-blocks.html#abi"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Function {
+                        name @output
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+
+                        importable_path {
+                            path @output @tag
+                        }
+                        
+                        span_: span @optional {
+                            filename @output
+                            begin_line @output
+                        }
+                        
+                        abi_: abi {
+                            name @output @tag
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on Function {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                        }
+                        
+                        abi @fold @transform(op: "count")  @filter(op: "=", value: ["$zero"]) {
+                            name @filter(op: "=", value: ["%name"])
+                        }
+
+                        new_abi_: abi {
+                            name @output
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "public": "public",
+        "zero": 0,
```

Comment:
```suggestion
```
I think we won't need this argument after applying the suggested tweak above.

Reply:
Thanks for the detailed feedback it was really useful.  That should be all the suggested changes done now.

Reply:
Happy to hear it! Thanks for working on this!